src/app/workflow/view/view.component.ts

```diff
@@ -89,6 +93,25 @@ export class ViewWorkflowComponent extends View implements OnInit {
     });
   }
 
+  snapshotVersion(): void {
+    // Create a new version temporarily with the frozen bit set. We assume the version is
+    // not already a snapshot since the UI controls shouldn't be available for
+    // snapshotted versions.
+    var snapshot = { ...this.version };
```

Comment:
This can be a `const`.

Also it's a shame this.version doesn't have a type, but it looks like it would take a bit of work to make it so and is outside the scope of this PR.

Reply:
For now I would suggest `const snapshot: WorkflowVersion =  {...this.version};`. Then your next line could be `workflowVersion.frozen =`, and in the extremely unlikely event the response model changes, the compiler would catch it.